enhancements/api-review/add-new-CRD-ImageSourceDigestPolicy-and-ImageSourceTagPolicy-to-config.openshift.io.md

```diff
@@ -0,0 +1,393 @@
+	// is by default considered lower priority than all mirrors. Other cluster configuration,
+	// including (but not limited to) other repositoryMirrors objects,
+	// may impact the exact order mirrors are contacted in, or some mirrors may be contacted
+	// in parallel, so this should be considered a preference rather than a guarantee of ordering.
+	// "mirrors" ueses one of the following formats:
+	// host[:port]
+	// host[:port]/namespace[/namespace…]
+	// host[:port]/namespace[/namespace…]/repo
+	// for more information about the format, see the document about the location field:
+	// https://github.com/containers/image/blob/main/docs/containers-registries.conf.5.md#choosing-a-registry-toml-table
+	// +optional
+	// +listType=set
+	Mirrors []Mirrors `json:"mirrors,omitempty"`
+}
+
+// +kubebuilder:validation:Pattern=`^((?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?)(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?$`
+type Mirror string
+```
+
+An example ImageSourceTagPolicy file will look like:
+```yaml
+apiVersion: config.openshift.io/v1
+kind: ImageSourceTagPolicy
+metadata:
+  name: ubi8repo
+spec:
+  repositoryTagMirrors: 
+  - mirrors:
+    - example.io/example/ubi-minimal
+    source: registry.access.redhat.com/ubi8/ubi-minimal
+```
+
+The ImageSourceTagPolicy object will lead to the creattion of a drop-in file at `/etc/containers/registries.conf` by MCO that no
+digest only limitations on this mirrors.
+
+### User Stories
+
+#### As a user, I would like to pull images from mirror using tag, without digests reference
+
+The user need to define multiple mirror configurations, used by apps/manifests that don't use digests to pull
+the images when working with disconnected environments or pulling from registries that act as transparent pull-through proxy cache.
+
+The user can pull image without digest by configuring mirrors using `ImageSourceTagPolicy` CRD.
+And MCO will create the ImageSourceTagPolicy object. Once this is done, the images can be pulled from the mirrors without the digest referenced.
+
+### API Extensions
+- Adds new CRD ImageSourceDigestPolicy to config.openshift.io/v1.
+- Adds new CRD ImageSourceTagPolicy to config.openshift.io/v1
+- Impacts on existing resources(The impacts are in the "Operational Aspects
+of API Extensions" section)
+  - Components use ImageContentSourcePolicy should upgrade to use ImageSourceDigestPolicy.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Implementing this enhancement requires changes in:
+- [containers/images](https://github.com/containers/image): registries.conf needs change its mirror setting to make digest only configurable for mirrors registry, not for the main registry. This can be achieved by having a boolean for each mirror table, or having separated mirror lists: a list of mirrors that allow for tag specification,
```

Comment:
```suggestion
- [containers/images](https://github.com/containers/image): registries.conf needs to change its mirror setting to make digests only configurable for mirrors registry, not for the main registry. This can be achieved by having a boolean for each mirror table, or having separated mirror lists: a list of mirrors that allow for tag specification,
```